frame/alliance/src/lib.rs

```diff
@@ -765,6 +702,60 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Disband the Alliance, remove all active members and unreserve deposits.
+		///
+		/// Witness data must be set.
+		#[pallet::weight(T::WeightInfo::disband(
+			witness.voting_members,
+			witness.ally_members,
+			witness.voting_members + witness.ally_members,
+		))]
+		pub fn disband(
+			origin: OriginFor<T>,
+			witness: DisbandWitness,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+
+			ensure!(!witness.is_zero(), Error::<T, I>::BadWitness);
```

Comment:
This check seems unnecessary given the following three?

Reply:
This would make sense if Alliance not set, and before failing on `is_initialized` check below it will do 3 reads. 
But we can fail earlier.